Lesson18.js

```diff
@@ -0,0 +1,189 @@
+//Pythagorean theorem
+//Calculate length of other side aa + bb = c*c
+//Find c
+
+//pythagorean(a, b);
+function pythagorean(a, b) {
+  return console.log(Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2)));
+}
+pythagorean(5, 12);
+// 13
+
+//formatMoney(num);
+function formatMoney(num) {
+  if (num <= 0) {
+    num = num.toFixed(2);
+  } else {
+    num = "+" + num.toFixed(2);
+  }
+  const parts = num.split(".");
+  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+  const formatMoney = parts.join(".");
+
+  return console.log(formatMoney);
+}
+
+formatMoney(1232323);
+//'+ 1,232,323.00'
+
+formatMoney(-23.2132);
+//'- 23.21'
+
+//Format number in spaces
+//+- sign should be present
+//Separate thousands with spaces
+
+//formatNumber(num);
+function formatNumber(value) {
+  let num = "";
+  if (value % 1 === 0) {
+    value = value.toString();
+  } else {
+    value = value.toFixed(3);
+  }
+  if (+value <= 0) {
+    num = value;
+  } else {
+    num = "+" + value;
+  }
+  const parts = num.split(".");
+  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, " ");
+  const formatNumber = parts.join(".");
+
+  return console.log(formatNumber);
+}
+
+formatNumber(1232323);
+// 1 232 323
+
+formatNumber(1223.65378);
+// 1 223.654
+//Write a password generator with length n
+//Generate password with any symbol
+//!"#$%&'()*+,-./0123456789:;<=>?
+//@ABCDEFGHIJKLMNOPQRSTUVWXYZ[]^_
+//`abcdefghijklmnopqrstuvwxyz
+
+function generatePassword(n) {
+  let charset =
+    "!\"#$%&'()*+,-./0123456789:;<=>?@[]^_?`abcdefghijklmnopqrstuvwxyzcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  let res = "";
+  for (let i = 0; i < n; i++) {
+    res += charset.charAt(Math.floor(Math.random() * charset.length));
+  }
+  return console.log(res);
```

Comment:
please don't return a console.log()
this doesn't make sense
![image](https://user-images.githubusercontent.com/32968019/221366291-2d32630c-9770-4a51-b26b-c1af9b2042a5.png)